### ISSUE TITLE
Quest: Supplying the Front (1578)

### ISSUE THREAD

**Issue by Freeman256** [TrinityCore/TrinityCore#26592]
https://classic.wowhead.com/quest=1578/supplying-the-front
https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?quest=1578

Description:
Quest not working. To complete the task, the player needs to have the blacksmithing profession raised by 35. If you bring the required items ([Copper Axe](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?item=2845) and [Copper Chain Belt](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?item=2851)) to [Thorvald Deepforge](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=6030), then the quest cannot be turned in. I tried restart the game with deleting the cache, dropping one of the required items and crafting it again - to no avail, restarting the server also didn't fix the problem. 

Expected behaviour:
After making the above items in the indicated quantities, the [Thorvald Deepforge](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=6030) must allow this quest to be completed.

Steps to reproduce the problem:
1. 	Create an Alliance character and raise the level of blacksmithing to 35. 
2.	Take the quest [Supplying the Front](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?quest=1578) from the [Tormus Deepforge](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=6031) in Ironforge.
3.	Craft the [Copper Axe](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?item=2845) and [Copper Chain Belt](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?item=2851) by 6.
4.	Take these items to [Thorvald Deepforge](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=6030) in the southern guard tower of Loch Modan.
5.	Make sure that, despite the conditions being met, the quest cannot be completed.
6.	Become sad.

Branch:
3.3.5

TC rev. hash/commit: https://github.com/TrinityCore/TrinityCore/commit/2acdd163fea281f467f074d267e45660fc9bf99d

TrinityCore rev. 2021-06-11 17:28:34 +0300 (3.3.5 branch) (Win64, Release, Static)

Some logs from auth/world servers:
Using SSL version: OpenSSL 1.1.1d 10 Sep 2019 (library: OpenSSL 1.1.1d 10 Sep 2019)
Using Boost version: 1.72.0
Using World DB: TDB 335.21051
MySQL client library: 5.7.28
MySQL server ver: 5.7.28
Operating system:
Win7 SP1 x64 with all updates (build 6.1.7601).

**Comment by Freeman256** [comment]
Well, I tried to remove all the required items from the inventory and make them for myself as a command **.additem**, and now the [Thorvald Deepforge](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=6030) is allowed to complete the quest. I don't understand why this is happening. Perhaps (but unlikely), it was influenced by the fact that trying to fix the problem, I started changing the values in the tables, namely:
UPDATE `world`.`quest_template` SET `QuestSortID`='1' WHERE  `ID`=1578;
Before the change, it had a value of **-121**. As I understand it, this column is responsible for sorting quests for characters, but I was alarmed by the negative value, which I changed to **1**. I'm not sure if this will completely fix the problem, so I will not close it. Suddenly, someone else is also lucky enough to face this problem.
